from launch_ros.actions import Node
from launch import LaunchDescription


def generate_launch_description():

    # Symlinks are auto-generated by V4L2
    # ls /dev/v4l/by-id to see all the available symlinks

    # USB 3.0 front (fisheye)
    front_cam_node: Node = Node(
        exec_name="front_cam",
        package="v4l2_camera",
        executable="v4l2_camera_node",
        namespace="front_cam",
        parameters=[
            {"video_device":
             "/dev/v4l/by-path/platform-fd500000.pcie-pci-0000:01:00.0-usb-0:4:1.0-video-index0"},
            {"image_size": [640, 480]}
        ],
        remappings=[("/pi/front_cam/image_raw", "/front_cam/image_raw")]
    )

    # USB 3.0 bottom
    bottom_cam_node: Node = Node(
        exec_name="bottom_cam",
        package="v4l2_camera",
        executable="v4l2_camera_node",
        namespace="bottom_cam",
        parameters=[
            {"video_device":
             "/dev/v4l/by-path/platform-fd500000.pcie-pci-0000:01:00.0-usb-0:1:1.0-video-index0"},
            {"image_size": [640, 480]}
        ],
        remappings=[("/pi/bottom_cam/image_raw", "/bottom_cam/image_raw")]
    )

    return LaunchDescription([
        front_cam_node,
        bottom_cam_node
    ])
